src/all/batoto/src/eu/kanade/tachiyomi/extension/all/batoto/BatoTo.kt

```diff
@@ -53,7 +53,11 @@ open class BatoTo(
     private val preferences by getPreferencesLazy { migrateMirrorPref() }
 
     override val name: String = "Bato.to"
-    override val baseUrl: String get() = mirror
+    override val baseUrl: String get() = when (System.getenv("CI")) {
+        "true" -> (MIRROR_PREF_ENTRY_VALUES.drop(1) + DEPRECATED_MIRRORS).joinToString("#, ")
```

Comment:
Put it inside `getMirrorPref()` as an early return.

I think you can also just `override var baseUrl` and absorb the `mirror` field.